### PR TITLE
Properly deserialize older blocks

### PR DESCRIPTION
**Pull request by hanssv**
If we want to be backwards compatible we should understand old blocks...

### CODE REVIEW

**Comment by hanssv** [comment]
Fighting dialyzer...

File: apps/aecore/src/aec_blocks.erl [hunk 1]
```diff
@@ -171,25 +171,32 @@ from_header_and_txs(#header{height = Height,
...
+            [{header, Hdr0}, {txs, Txs0}] =
+                aec_object_serialization:deserialize(block, Vsn, Template, Bin),
+            Hdr = aec_headers:deserialize_from_binary(Hdr0),
+            Txs = [ aetx_sign:deserialize_from_binary(Tx) || Tx <- Txs0 ],
+            {ok, from_header_and_txs(Hdr, Txs)};
+        Err = {error, _} ->
+            Err
+    end.
+
+serialization_template(Vsn) when Vsn >= 9 andalso Vsn =< ?PROTOCOL_VERSION ->
```
Comment (lucafavatella) [inline apps/aecore/src/aec_blocks.erl]: 9 -> macro?

File: apps/aecore/src/aec_blocks.erl [hunk 2]
```diff
@@ -171,25 +171,32 @@ from_header_and_txs(#header{height = Height,
 serialize_to_binary(B = #block{}) ->
     Hdr = aec_headers:serialize_to_binary(to_header(B)),
     Txs = [ aetx_sign:serialize_to_binary(Tx) || Tx <- B#block.txs ],
+    Vsn = B#block.version,
+    {ok, Template} = serialization_template(Vsn),
     aec_object_serialization:serialize(
         block,
-        ?PROTOCOL_VERSION,
-        serialization_template(?PROTOCOL_VERSION),
+        Vsn,
```
Comment (lucafavatella) [inline apps/aecore/src/aec_blocks.erl]: Much needed indeed.

**Comment by hanssv** [comment]
Don't see the same issue locally:
```
===> Running Common Test suites...
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.start_first_node: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.test_subscription: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.mine_on_first: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.start_second_node: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.start_third_node: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.mine_again_on_first: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.mine_on_second: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.mine_on_third: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.restart_third: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.report_metrics: OK
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.check_metrics_logged: OK
All 11 tests passed.
```

Will restart the tests

**Comment by lucafavatella** [comment]
I will merge soon.

**Comment by lucafavatella** [comment]
Same as #973

**Pull request merged**